large_image/tilesource/base.py

```diff
@@ -128,12 +128,15 @@ def __init__(self, encoding='JPEG', jpegQuality=95, jpegSubsampling=0,
         self.edge = edge
         self._jsonstyle = style
```

Comment:
We use this as part of the caching key.  If this is a dict, then it has to serialize the dict each time.  I think the resolution would be to change
```
            if isinstance(style, dict):
                self.style = style
```
add
```
            if isinstance(style, dict):
                self.style = copy.deepcopy(style)
                self._jsonstyle = json.dumps(style, sort_keys=True, separators=(',', ':'))
```
I've added a deepcopy to the dictionary, as otherwise the caller could alter the style dynamically which would probably break things.

Reply:
Push a fix for this